sensu/plugins/metrics-ceph-pool-capacity.rb

```diff
@@ -0,0 +1,145 @@
+#! /usr/bin/env ruby
+#
+# metrics-ceph-pool-capacity
+#
+# DESCRIPTION:
+#   This plugin is used to get ceph capacity metrics for using ceph as
+#   OpenStack Cinder's driver. 
+#   
+#   *.poolname.used_bytes   using 'ceph df' to get 'bytes_used'
+#   *.poolname.max_avail_bytes   using 'ceph df' to get 'max_avail'
+#   *.poolname.total_bytes using 'ceph df' to get 'bytes_used'+'max_avail'
+#   *.poolname.cinder_allocated_bytes  using 'rbd ls -l' to sum the 'SIZE'
+#   *.poolname.cinder_avail_bytes  total_bytes - cinder_allocated_bytes
+#   (suppose this ceph is only used for openstack cinder)
+#
+#   since ceph support thin provision, total_allocated_bytes >= total_used_bytes
+#
+# OUTPUT:
+#   metrics data in Graphite's format
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   ceph client
+#
+# USAGE:
+#   #YELLOW
+#
+# NOTES:
+#   Runs 'ceph df' and 'rbd ls -l' command(s) to report usage of ceph cluster.  
+#   May need read access to ceph keyring and/or root access for 
+#   authentication.
+#
+# LICENSE:
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-plugin/metric/cli'
+require 'timeout'
+require 'English'
+require 'json'
+require 'socket'
+
+class CephPoolCapacity < Sensu::Plugin::Metric::CLI::Graphite
+
+  option :scheme,
+    :description => "Metric naming scheme, text to prepend to .$parent.$child",
+	:short => "-s SCHEME",
+    :long => "--scheme SCHEME",
+    :default => "#{Socket.gethostname}.usage.ceph"
```

Comment:
this should be `"#{Socket.gethostname}.ceph"`   usage is pretty meaningless in a prefix.